jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java

```diff
@@ -120,6 +145,52 @@ private static DiagnosticProperties getHeapStatistics(String diagnosticCommand)
 		}
 		return DiagnosticProperties.makeStringResult(result);
 	}
+	
+	private static DiagnosticProperties getThreadInfo(String diagnosticCommand) {
+		DiagnosticProperties result = null;
+		boolean okay = true;
+		boolean addSynchronizers = false;
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		/* only one option ("-l[=<BOOLEAN>]") is allowed, so the diagnosticCommand can comprise at most
+		 * the base command and one option, with an optional value */
+		if (parts.length > 2) {
+			okay = false;
+		} else if (parts.length == 2) {
+			String option = parts[1];
+			if (option.startsWith(THREAD_LOCKED_SYNCHRONIZERS_OPTION)) {
+				if ((THREAD_LOCKED_SYNCHRONIZERS_OPTION.length() == option.length()) /* exact match */
+						|| option.toLowerCase().endsWith("=true")) { //$NON-NLS-1$
+					addSynchronizers = true;
+				}
+			} else {
+				okay = false;
+			}
+		}
+		if (!okay) {
+			result = DiagnosticProperties.makeErrorProperties("Command not recognized: " + diagnosticCommand); //$NON-NLS-1$
+		} else {
+			PrintWriter p = new PrintWriter(new StringWriter(2000));
```

Comment:
This will not work: `PrintWriter.toString()` does not return the printed text. You need to use `StringWriter.toString()` which means you need a local variable to access that object, and you should call `PrintWriter.flush()` before that.

Please also improve the local variable names: one or two letters is insufficient.